examples/hotrod/pkg/tracing/init.go

```diff
@@ -43,6 +44,40 @@ import (
 
 var once sync.Once
 
+// InitOTEL initializes OpenTelemetry SDK and uses OTel-OpenTracing Bridge
+// to return an OpenTracing-compatible tracer.
+func InitOTEL(serviceName string, exporterType string, metricsFactory metrics.Factory, logger log.Factory) trace.Tracer {
+	once.Do(func() {
+		otel.SetTextMapPropagator(
+			propagation.NewCompositeTextMapPropagator(
+				propagation.TraceContext{},
+				propagation.Baggage{},
+			))
+	})
+
+	exp, err := createOtelExporter(exporterType)
+	if err != nil {
+		logger.Bg().Fatal("cannot create exporter", zap.String("exporterType", exporterType), zap.Error(err))
+	}
+	logger.Bg().Debug("using " + exporterType + " trace exporter")
+
+	rpcmetricsObserver := rpcmetrics.NewObserver(metricsFactory, rpcmetrics.DefaultNameNormalizer)
+
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exp),
+		sdktrace.WithSpanProcessor(rpcmetricsObserver),
+		sdktrace.WithResource(resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(serviceName),
+		)),
+	)
+	otel.SetTracerProvider(tp)
+	// _, otelTracerProvider := otbridge.NewTracerPair(tp.Tracer(""))
+	tracer := otel.Tracer("my otel tracer")
+	// logger.Bg().Debug("created OTEL->OT bridge", zap.String("service-name", serviceName))
+	return tracer
+}
```

Comment:
all of this is already done in the Init function below. You can simply change that function to return two objects:
```
func Init() (opentracing.Tracer, trace.TracerProvider) {}
```

Reply:
Or at least refactor in such a way that we don't duplicate the code, e.g. 
```
func InitOTEL() trace.TracerProvider {
  _, otel := initBoth()
  return otel
}
// rename Init() to initBoth()
func initBoth() (opentracing.Tracer, trace.TracerProvider) {}
```

Reply:
Ah thanks, that'll be nice.